README.rst

```diff
@@ -0,0 +1,158 @@
+
+#. Run the script **manage_externals/checkout_externals.py**. ::
+
+      > ./manage_externals/checkout_externals.py
+
+   The **checkout_externals.py** script is a package manager that will
+   populate the cesm directory with the relevant versions of each of the
+   components along with the CIME infrastructure code.
+
+At this point you have a working version of CESM.
+
+To see full details of how to set up a case, compile and run, see the CIME documentation at http://esmci.github.io/cime/ .
+
+More details on checkout_externals.py
+-------------------------------------
+
+The file **CESM.cfg** in your top-level CESM directory tells
+**checkout_externals.py** which tag/branch of each component should be
+brought in to generate your sandbox. (This file serves the same purpose
+as SVN_EXTERNAL_DIRECTORIES when CESM was in a subversion repository.)
+
+**You need to rerun checkout_externals.py whenever CESM.cfg has
+changed** (unless you have already manually updated the relevant
+external(s) to have the correct branch/tag checked out). Common times
+when this is needed are:
+
+* After checking out a new CESM branch/tag
+
+* After merging some other CESM branch/tag into your currently
+  checked-out branch
+
+**checkout_externals.py** must be run from the root of the source tree -
+i.e., your current working directory must be the directory containing
+**CESM.cfg**.
+
+To see more details of **checkout_externals.py**, issue ::
+
+  > ./manage_externals/checkout_externals.py --help
+
+Customizing your CESM sandbox
+=============================
+
+There are several use cases to consider when you want to customize or modify your CESM sandbox.
+
+Switching to a different CESM tag
+---------------------------------
+
+If you have already checked out a tag and **HAVE NOT MADE ANY
+MODIFICATIONS** it is simple to change your sandbox. Say that you
+checked out cesm2.0.beta07 but really wanted to have cesm2.0.beta08;
+you would simply do the following::
+
+  > git checkout cesm2.0.beta08
+  > ./manage_externals/checkout_externals.py
+
+You should **not** use this method if you have made any source code
+changes, or if you have any ongoing CESM cases that were created from
+this sandbox. In these cases, it is often easiest to do a second **git
+clone**.
+
+Pointing to a different version of a component
+----------------------------------------------
+
+Each entry in **CESM.cfg** has the following form (we use CAM as an
+example below)::
+ 
+  [cam]
+  tag = trunk_tags/cam5_4_143/components/cam
+  protocol = svn
+  repo_url = https://svn-ccsm-models.cgd.ucar.edu/cam1
+  local_path = components/cam
+  required = True
+
+Each entry specifies either a tag or a branch. To point to a new tag:
+
+#. Modify the relevant entry/entries in **CESM.cfg** (e.g., changing
+   ``cam5_4_143`` to ``cam5_4_144`` above)
+
+#. Checkout the new component(s)::
+
+     > ./manage_externals/checkout_externals.py
+
+Committing your change to CESM.cfg
```

Comment:
Do we want to encourage people to modify CESM.cfg? Or do we want to encourage people to create their own myCESM.cfg file and leave CESM.cfg alone?

Advantages of myCESM.cfg:
* no merge conflicts to resolve.
* can return to original version without switching branches

Disadvantages af myCESM.cfg:
* must add '--model myCESM.cfg' to the checkout_externals commandline.

Reply:
I feel that modifying CESM.cfg will be a better workflow in general. This is largely because I actually see your advantage of myCESM.cfg, "no merge conflicts to resolve", as a disadvantage: In the scenario where someone points to a branch of a component and then wants to update their sandbox to a newer CESM beta tag, I think it will be better if they have modified CESM.cfg and thus are notified of possible conflicts, rather than having the merge to the latest CESM beta tag effectively accomplish nothing.

That said, I don't feel strongly about this and could be convinced otherwise if others feel we should recommend using myCESM.cfg.